pkg/secrets/secrets.go

```diff
@@ -21,3 +40,44 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return errors.Errorf("Unsupported type '%s' for secret '%s:%s'", string(secret.Type), secret.Namespace, secret.Name)
 	}
 }
+
+func getLocationType(secret *v1.Secret) (repositoryserver.RepositoryServerSecrets, error) {
+	var locationType []byte
+	var ok bool
+	if secret == nil {
+		return nil, errors.New("Secret is Nil")
+	}
+
+	if locationType, ok = (secret.Data[LocationTypeKey]); !ok {
+		return nil, errors.Errorf("secret '%s:%s' does not have required field %s", secret.Namespace, secret.Name, LocationTypeKey)
+	}
+
+	switch string(locationType) {
+	case AWSLocationSecretType:
```

Comment:
We'll need to validate other location types as well. See https://github.com/kanisterio/kanister/blob/f74570f64b56527abd7e6440a1b8a328465497ce/pkg/kopia/command/storage/secret_utils.go#L41-L45.
We can probably re-use the LocType type here.